stock-filters/ChirpVillage/YardGenerator.py

```diff
@@ -150,7 +150,7 @@ def init_surface(self, new_surface, x_start, x_end, z_start, z_end):
         for i in range(x_start, x_end):
             for j in range(z_start, z_end):
                 block_probability = float((lehmer_2(lehmer_2(self.seed, i), j)) % 100) / 100
-                if block_probability < probability_spawning_yard_block:
+                if block_probability < probability_spawning_yard_block and not new_surface.surface_map[i][j].is_water and not new_surface.surface_map[i][j].is_lava:
```

Comment:
good point, we shouldnt have yards on water

Reply:
Yee, glad it was so easy to fix too!